packages/bot/src/stores/toolbar-store.js

```diff
@@ -0,0 +1,269 @@
+        Blockly.Events.setGroup('reset');
+        workspace.clear();
+        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(workspace.blocksXmlStr), workspace);
+        Blockly.Events.setGroup(false);
+
+        this.file_name = translate('Untitled Bot');
+    }
+
+    @action.bound
+    toggleSaveLoadModal(is_save) {
+        this.is_saveload_modal_open = !this.is_saveload_modal_open;
+        this.is_save_modal = is_save;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onDriveConnect() {
+        googleDrive.authorise().then(() => {
+            // TODO
+        });
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onLoadClick({ is_local }) {
+        if (is_local) {
+            const upload = document.getElementById('files');
+            upload.click();
+        } else {
+            // TO DO
+        }
+    }
+
+    @action.bound
+    handleFileChange(e) {
+        let files,
+            drop_event;
+        if (e.type === 'drop') {
+            e.stopPropagation();
+            e.preventDefault();
+            ({ files } = e.dataTransfer);
+            drop_event = e;
+        } else {
+            ({ files } = e.target);
+            this.toggleSaveLoadModal();
+        }
+        files = Array.from(files);
+        files.forEach(file => {
+            this.file_name = file.name.replace(/\.[^/.]+$/, '');
+
+            if (file.type.match('text/xml')) {
+                this.readFile(file, drop_event);
+            } else {
+                // TODO
+                console.error('File Type not matched'); // eslint-disable-line
+            }
+        });
+        e.target.value = '';
+    }
+
+    @action.bound
+    onConfirmSave(values) {
+        const { is_local, save_as_collection } = values;
+
+        if (is_local) {
+            const xml = Blockly.Xml.workspaceToDom(Blockly.derivWorkspace);
+            xml.setAttribute('collection', save_as_collection ? 'true' : 'false');
+
+            const data = Blockly.Xml.domToPrettyText(xml);
+            const blob = new Blob([data], { type: 'text/xml;charset=utf-8' });
+
+            filesaver.saveAs(blob, `${this.file_name}.xml`);
+        } else {
+            // TO DO
+        }
+
+        this.toggleSaveLoadModal();
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onUndoClick() {
+        Blockly.Events.setGroup('undo');
+        Blockly.derivWorkspace.undo();
+        Blockly.Events.setGroup(false);
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onRedoClick() {
+        Blockly.derivWorkspace.undo(true);
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onZoomInOutClick(is_zoom_in) {
+        const metrics = Blockly.derivWorkspace.getMetrics();
+        const addition = is_zoom_in ? 1 : -1;
+
+        Blockly.derivWorkspace.zoom(metrics.viewWidth / 2, metrics.viewHeight / 2, addition);
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    onSortClick() {
```

Comment:
Why not call `cleanUp` on the workspace instance see `workspace_svg.js`? We have our own way of sorting blocks, so don't re-implement the vanilla way of rearranging blocks.